src/status_im/contexts/keycard/change_pin/events.cljs

```diff
@@ -0,0 +1,80 @@
+            [utils.re-frame :as rf]))
+
+(rf/reg-event-fx :keycard/change-pin.enter-current-pin
+ (fn [_]
+   {:fx [[:dispatch [:navigate-back]]
+         [:dispatch
+          [:open-modal :screen/keycard.pin.enter
+           {:title       (i18n/label :t/enter-current-keycard-pin)
+            :on-complete (fn [current-pin]
+                           (rf/dispatch [:navigate-back])
+                           (rf/dispatch [:keycard/change-pin.save-current-pin current-pin])
+                           (rf/dispatch [:keycard/change-pin.enter-new-pin]))}]]]}))
+
+(rf/reg-event-fx :keycard/change-pin.save-current-pin
+ (fn [{:keys [db]} [pin]]
+   {:db (assoc-in db [:keycard :change-pin :current-pin] pin)}))
+
+(rf/reg-event-fx :keycard/change-pin.enter-new-pin
+ (fn [_]
+   {:fx [[:dispatch
+          [:open-modal :screen/keycard.pin.create
+           {:changing-pin? true
+            :on-complete   (fn [new-pin]
+                             (rf/dispatch [:navigate-back])
+                             (rf/dispatch [:keycard/change-pin.save-new-pin new-pin])
+                             (rf/dispatch [:open-modal :screen/keycard.ready-to-change-pin]))}]]]}))
+
+(rf/reg-event-fx :keycard/change-pin.save-new-pin
+ (fn [{:keys [db]} [pin]]
+   {:db (assoc-in db [:keycard :change-pin :new-pin] pin)}))
+
+(rf/reg-event-fx :keycard/change-pin.verify-current-pin-and-continue
+ (fn [{:keys [db]}]
+   {:fx [[:dispatch
+          [:keycard/connect
+           {:on-error
+            (fn [error]
+              (if (or (= error :keycard/error.keycard-blank)
+                      (= error :keycard/error.keycard-wrong-profile))
+                (rf/dispatch
+                 [:keycard/verify-pin
+                  {:pin        (get-in db [:keycard :change-pin :current-pin])
+                   :on-success (fn []
+                                 (rf/dispatch [:open-modal :screen/keycard.changing-keycard-pin])
+                                 (rf/dispatch [:keycard/change-pin.change-pin-and-continue]))
+                   :on-failure (fn [error]
+                                 (rf/dispatch [:keycard/disconnect])
+                                 (rf/dispatch [:keycard/on-action-with-pin-error error])
+                                 (rf/dispatch [:keycard/change-pin.enter-current-pin]))}])
+                (rf/dispatch [:keycard/on-application-info-error error])))}]]]}))
+
+(rf/reg-event-fx :keycard/change-pin.change-pin-and-continue
+ (fn [{:keys [db]}]
+   {:fx [[:dispatch
+          [:keycard/update-event-vector-and-dispatch
+           {:on-error
+            (fn [error]
+              (if (or (= error :keycard/error.keycard-blank)
+                      (= error :keycard/error.keycard-wrong-profile))
+                (rf/dispatch
+                 [:keycard/change-pin
+                  {:on-success  (fn []
+                                  (rf/dispatch [:navigate-back])
+                                  (rf/dispatch [:navigate-back])
```

Comment:
Initially, instead of closing both modals at the same time, I closed the first modal when navigating to the second. However, since changing the pin is quite fast, it caused an awkward transition (see at 00:13 in [this video](https://github.com/user-attachments/assets/985ba6ed-d6a7-494f-97b7-85111313311a)). Now, I keep both modals open and close them simultaneously once the pin is changed.